test/metric_value_benchmark/collectd_test.go

```diff
@@ -72,41 +59,58 @@ func (t *CollectDTestRunner) validateCollectDMetric(metricName string) status.Te
 		Status: status.FAILED,
 	}
 
-	dims, failed := t.DimensionFactory.GetDimensions([]dimension.Instruction{
+	instructions := []dimension.Instruction{
 		{
 			Key:   "InstanceId",
 			Value: dimension.UnknownDimensionValue(),
 		},
-		{
-			Key:   "type_instance",
-			Value: dimension.ExpectedDimensionValue{aws.String("user")},
-		},
-		{
-			Key:   "instance",
-			Value: dimension.ExpectedDimensionValue{aws.String("0")},
-		},
-		{
+	}
+	switch metricName {
+	case "collectd_counter_1_value":
+		instructions = append(instructions, dimension.Instruction{
+			// CWA adds this metric_type dimension.
 			Key:   "type",
-			Value: dimension.ExpectedDimensionValue{aws.String("percent")},
-		},
-	})
+			Value: dimension.ExpectedDimensionValue{Value: aws.String("counter")},
+		})
+	case "collectd_gauge_1_value":
+		instructions = append(instructions, dimension.Instruction{
+			// CWA adds this metric_type dimension.
+			Key:   "type",
+			Value: dimension.ExpectedDimensionValue{Value: aws.String("gauge")},
+		})
+	}
 
+	dims, failed := t.DimensionFactory.GetDimensions(instructions)
 	if len(failed) > 0 {
 		return testResult
 	}
-
 	fetcher := metric.MetricValueFetcher{}
+	// Namespace must match the JSON config.
```

Comment:
This is a given. We don't need this comment. It should have been caught at review time before.

Reply:
Got cha. Thanks for your comments on it !!!